engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java

```diff
@@ -183,6 +198,28 @@ public Optional<Long> findPreviousVersionDecisionKey(
     }
   }
 
+  private Optional<Long> findPreviousVersionDecisionRequirementsKey(
+      final DirectBuffer decisionRequirementsId, final int currentVersion) {
+    final Map<Integer, Long> decisionRequirementsKeysByVersion = new HashMap<>();
+
+    dbDecisionRequirementsId.wrapBuffer(decisionRequirementsId);
+    decisionRequirementsKeyByIdAndVersion.whileEqualPrefix(
+        dbDecisionRequirementsId,
+        ((key, drgKey) -> {
+          if (key.second().getValue() < currentVersion) {
+            decisionRequirementsKeysByVersion.put(
+                key.second().getValue(), drgKey.inner().getValue());
+          }
+        }));
+
+    if (decisionRequirementsKeysByVersion.isEmpty()) {
+      return Optional.empty();
+    } else {
+      final Integer previousVersion = Collections.max(decisionRequirementsKeysByVersion.keySet());
+      return Optional.of(decisionRequirementsKeysByVersion.get(previousVersion));
+    }
+  }
+
```

Comment:
Note that this is untested and unused code for now. Before it was public and I had tests for it, but I refactored it to be private. It will be thoroughly tested as part of #9773, when we will test the deletion of DRGs.

I chose to keep the method in this PR as it shows the intention of the new ColumnFamily. I also didn't want to rewrite it 😄